### PR TITLE
fcm test-battery: cm: adjust for Subversion 1.9 compatibility

### PR DESCRIPTION
**Pull request by benfitzpatrick**
This allows `fcm test-battery` to be run successfully under Subversion 1.9 (1.9.4 tested) as well
as the existing support for Subversion 1.8.

The test-battery changes are mainly due to changes in `diff` and `mergeinfo` output at 1.9.

### CODE REVIEW

File: t/fcm-merge/test_header [hunk 1]
```diff
@@ -224,13 +224,26 @@ function test_mergeinfo() {
...
+    local WC_REVISION=${4:-WC}
+    if "$SVN_VERSION_IS_19"; then
+        WC_REVISION=WC
+    elif (( WC_REVISION < 10 )); then
+        WC_REVISION="$WC_REVISION "
+    fi
     local RESULTS_FILE=$(mktemp)
-    cat "$INPUT_RESULTS_FILE" > "$RESULTS_FILE"
+    cat "$INPUT_RESULTS_FILE" >"$RESULTS_FILE"
+    sed -i "s/^\(.\{23\}\)WC/\1$WC_REVISION/" "$RESULTS_FILE"
```
Comment (matthewrmshin) [inline t/fcm-merge/test_header]: Can combine the above 2 statements in a single statement:

```sh
sed "s/^\(.\{23\}\)WC/\1$WC_REVISION/" "$INPUT_RESULTS_FILE" >"$RESULTS_FILE"
```

File: t/fcm-merge/test_header [hunk 2]
```diff
@@ -224,13 +224,26 @@ function test_mergeinfo() {
...
     # Test svn:mergeinfo property.
     TEST_KEY=$TEST_INFO_NAME-prop
-    run_pass "$TEST_KEY" svn propget svn:mergeinfo .
-    file_cmp "$TEST_KEY.out" "$TEST_KEY.out" <$(get_results prop $RESULTS_FILE)
-    file_cmp "$TEST_KEY.err" "$TEST_KEY.err" </dev/null
+    PROP_RESULT=$(cat <$(get_results prop $RESULTS_FILE))
```
Comment (matthewrmshin) [inline t/fcm-merge/test_header]: Can just be this?

```sh
PROP_RESULT=$(get_results prop "$RESULTS_FILE")
```

**Comment by benfitzpatrick** [comment]
Addressed feedback (or made the PROP_RESULT bit clearer anyway). Rebased so we can have a Travis check.

**Comment by matthewrmshin** [comment]
Unfortunately, there appears to be a sort order issue on Travis CI. Good otherwise.

**Pull request merged**